src/test/java/com/faforever/client/login/LoginControllerTest.java

```diff
@@ -76,4 +85,15 @@ public void testForgotPasswordButtton() throws Exception {
 
     verify(platformService).showDocument("forgot");
   }
+
+  @Test
+  public void testUsernameEmailWarning() {
+    instance.usernameInput.setText("test@example.com");
+    instance.passwordInput.setText("foo");
+    instance.loginButton.fire();
+    verify(i18n).get(LOGIN_WITH_EMAIL_WARNING_KEY);
+    verifyZeroInteractions(userService);
+    assertThat(instance.loginErrorLabel.isVisible(), is(true));
+    assertThat(instance.loginErrorLabel.getText(), is(LOGIN_WITH_EMAIL_WARNING_KEY));
```

Comment:
In positive case we should assert that the loginErrorLabel is not visible.

Reply:
Well this not really his job tho

Reply:
What? If the code is wrong and always shows the error message? how do you test it then?